### ISSUE TITLE
Get Overflow when using DeepSpeed+fp16.

### ISSUE THREAD

**Issue by OleksandrKorovii** [huggingface/accelerate#992]
### System Info

```Shell
- `Accelerate` version: 0.15.0
- Platform: Linux-5.15.0-58-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- Numpy version: 1.24.1
- Transformers version: 4.25.1
- DeepSpeed version: try both 0.7.7 and 0.8.0
- PyTorch version (GPU?): 1.13.1+cu117 (True)
- `Accelerate` default config:
	- compute_environment: LOCAL_MACHINE
	- distributed_type: DEEPSPEED
	- mixed_precision: fp16
	- use_cpu: False
	- dynamo_backend: NO
	- num_processes: 2
	- machine_rank: 0
	- num_machines: 1
	- gpu_ids: None
	- main_process_ip: None
	- main_process_port: None
	- rdzv_backend: static
	- same_network: True
	- main_training_function: main
	- deepspeed_config: {'gradient_accumulation_steps': 1, 'offload_optimizer_device': 'cpu', 'offload_param_device': 'cpu', 'zero3_init_flag': False, 'zero_stage': 2}
	- fsdp_config: {}
	- megatron_lm_config: {}
	- downcast_bf16: no
	- tpu_name: None
	- tpu_zone: None
	- command_file: None
	- commands: None
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

I run the [official script](https://github.com/huggingface/transformers/blob/main/examples/pytorch/language-modeling/run_clm_no_trainer.py) for training GPT2-medium on my own dataset. During training I get followed error: 

```
[2023-01-20 09:18:06,767] [INFO] [stage3.py:1829:_overflow_clean_up] [deepspeed] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 4294967296, reducing to 2147483648.0
```

Command to run script:
```
    accelerate launch train.py \
          --train_file=accelarate_train.json \
          --model_name_or_path=microsoft/DialoGPT-medium \
          --output_dir=output_accelerate_1 \
          --per_device_train_batch_size=5 \
          --per_device_eval_batch_size=5  \
          --with_tracking \
          --overwrite_cache \
          --num_train_epochs=5 \
          --checkpointing_steps=epoch \
          --report_to=wandb
```

### Expected behavior

```Shell
script launched in mixed precision(fp16) without overflow
```

**Comment by pacman100** [comment]
Hello @OleksandrKorovii, it isn't an error, **those are warnings that can safely be ignored**. The optimizer and scheduler step is just skipped because of large scaling factor which lead to gradient overflow.  

If you want to control the initial scaling factor, use the deepspeed config file as given here in the docs: https://huggingface.co/docs/accelerate/usage_guides/deepspeed#deepspeed-config-file. You need to lower the `initial_scale_power` as per your liking in order to avoid those initial warnings.

**Comment by OleksandrKorovii** [comment]
@pacman100 I thought this the warning but, in the result script fails with the Exception

```
Exception: Current loss scale already at minimum - cannot decrease scale anymore. Exiting run.
```

**Comment by pacman100** [comment]
Hello @OleksandrKorovii, in that case, something else is wrong, can you check if the loss is `NaN`? I faced such an issue once wherein data loading was incorrect. 

Can you please provide the script and the command that you are using so that we can reproduce this issue?

**Comment by OleksandrKorovii** [comment]
I've checked the loss on first steps `Loss := 10.3359375`, it's not `NaN`

Script is [here](https://gist.github.com/OleksandrKorovii/c592a2929da5197d52fb39e13a39d3ad) 

Command to launch script
```
    accelerate launch train.py \
          --train_file=accelarate_train.json \
          --model_name_or_path=microsoft/DialoGPT-medium \
          --output_dir=output_accelerate_1 \
          --per_device_train_batch_size=5 \
          --per_device_eval_batch_size=5  \
          --with_tracking \
          --overwrite_cache \
          --num_train_epochs=5 \
          --checkpointing_steps=epoch \
          --report_to=wandb
```


I used json my own dataset with following format 
```
{
"version": "0.0.1",
"data":  [
    {"text": "Long text1"}, 
    {"text": "Long textN"},
]
}
```
